diff --git a/PhysicsAnalysis/SUSYPhys/SUSYTools/Root/Jets.cxx b/PhysicsAnalysis/SUSYPhys/SUSYTools/Root/Jets.cxx
index dbf3ea0abb..98a39a39c0 100644
--- a/PhysicsAnalysis/SUSYPhys/SUSYTools/Root/Jets.cxx
+++ b/PhysicsAnalysis/SUSYPhys/SUSYTools/Root/Jets.cxx
@@ -432,7 +432,7 @@ namespace ST {
 
   bool SUSYObjDef_xAOD::IsBadJet(const xAOD::Jet& input) const {
 
-    if ( !acc_passOR(input) ) return false;
+    //if ( !acc_passOR(input) ) return false;
 
     float ptcut = 20e3;
     bool  isPileup = !acc_passJvt(input);
@@ -569,7 +569,8 @@ namespace ST {
     ConstDataVector<xAOD::JetContainer> jvtjets(SG::VIEW_ELEMENTS);
     for (const auto& jet : *jets) {
       // Only jets that were good for every cut except JVT
-      if (acc_signal_less_JVT(*jet) && acc_passOR(*jet)) {
+      //if (acc_signal_less_JVT(*jet) && acc_passOR(*jet)) {
+      if(true) { // dantrim
         jvtjets.push_back(jet);
       }
     }
diff --git a/PhysicsAnalysis/SUSYPhys/SUSYTools/Root/SUSYObjDef_xAOD.cxx b/PhysicsAnalysis/SUSYPhys/SUSYTools/Root/SUSYObjDef_xAOD.cxx
index 16e2e17994..cd99b818a2 100644
--- a/PhysicsAnalysis/SUSYPhys/SUSYTools/Root/SUSYObjDef_xAOD.cxx
+++ b/PhysicsAnalysis/SUSYPhys/SUSYTools/Root/SUSYObjDef_xAOD.cxx
@@ -2120,11 +2120,11 @@ float SUSYObjDef_xAOD::GetDataWeight(const std::string& trig) {
   return m_prwTool->getDataWeight( *evtInfo, trig );
 }
 
-float SUSYObjDef_xAOD::GetCorrectedAverageInteractionsPerCrossing() {
+float SUSYObjDef_xAOD::GetCorrectedAverageInteractionsPerCrossing(bool includeDataSF) {
 
   const xAOD::EventInfo* evtInfo = 0;
   ATH_CHECK( evtStore()->retrieve( evtInfo, "EventInfo" ) );
-  return m_prwTool->getCorrectedMu( *evtInfo );
+  return m_prwTool->getCorrectedMu( *evtInfo, includeDataSF );
 }
 
 double SUSYObjDef_xAOD::GetSumOfWeights(int channel) {
diff --git a/PhysicsAnalysis/SUSYPhys/SUSYTools/SUSYTools/ISUSYObjDef_xAODTool.h b/PhysicsAnalysis/SUSYPhys/SUSYTools/SUSYTools/ISUSYObjDef_xAODTool.h
index ca2f719087..c889c2e565 100644
--- a/PhysicsAnalysis/SUSYPhys/SUSYTools/SUSYTools/ISUSYObjDef_xAODTool.h
+++ b/PhysicsAnalysis/SUSYPhys/SUSYTools/SUSYTools/ISUSYObjDef_xAODTool.h
@@ -352,7 +352,7 @@ namespace ST {
 
     virtual float GetDataWeight(const std::string&) = 0;
  
-    virtual float GetCorrectedAverageInteractionsPerCrossing() = 0;
+    virtual float GetCorrectedAverageInteractionsPerCrossing(bool includeDataSF) = 0;
 
     virtual double GetSumOfWeights(int channel) = 0;
     
diff --git a/PhysicsAnalysis/SUSYPhys/SUSYTools/SUSYTools/SUSYObjDef_xAOD.h b/PhysicsAnalysis/SUSYPhys/SUSYTools/SUSYTools/SUSYObjDef_xAOD.h
index 276d883158..c9d0a16d5f 100644
--- a/PhysicsAnalysis/SUSYPhys/SUSYTools/SUSYTools/SUSYObjDef_xAOD.h
+++ b/PhysicsAnalysis/SUSYPhys/SUSYTools/SUSYTools/SUSYObjDef_xAOD.h
@@ -288,7 +288,7 @@ namespace ST {
 
     float GetDataWeight(const std::string&) override final;
 
-    float GetCorrectedAverageInteractionsPerCrossing() override final;
+    float GetCorrectedAverageInteractionsPerCrossing(bool includeDataSF=false) override final;
 
     double GetSumOfWeights(int channel) override final;
 
