Index: SUSYTools/Root/SUSYObjDef_xAOD.cxx
===================================================================
--- SUSYTools/Root/SUSYObjDef_xAOD.cxx	(revision 743384)
+++ SUSYTools/Root/SUSYObjDef_xAOD.cxx	(working copy)
@@ -86,6 +86,10 @@
 static SG::AuxElement::Decorator<double> dec_effscalefact("effscalefact");
 static SG::AuxElement::Decorator<double> dec_jvtscalefact("jvtscalefact");
 
+static SG::AuxElement::Decorator<char> dec_photonPassCleaning("passCleaning");
+static SG::AuxElement::Decorator<char> dec_photonPassAmbiguity("passAmbiguity");
+
+
 SUSYObjDef_xAOD::SUSYObjDef_xAOD( const std::string& name )
   : asg::AsgMetadataTool( name ),
     m_dataSource(Undefined),
@@ -1649,6 +1653,9 @@
   dec_baseline(input) = false;
   dec_isol(input) = false;
 
+  dec_photonPassCleaning(input) = false;
+  dec_photonPassAmbiguity(input) = false;
+
   // Author cuts needed according to https://twiki.cern.ch/twiki/bin/view/AtlasProtected/EGammaIdentificationRun2#Photon_authors
   if ( !(input.author() & (xAOD::EgammaParameters::AuthorPhoton + xAOD::EgammaParameters::AuthorAmbiguous)) )
     return StatusCode::SUCCESS;
@@ -1673,7 +1680,10 @@
   float ph_rphi = input.showerShapeValue(xAOD::EgammaParameters::Rphi);
 
   if ( (ph_OQ & 134217728) != 0 && (ph_reta > 0.98 || ph_rphi > 1.0 || (ph_OQ & 67108864) != 0) ) return StatusCode::SUCCESS;
+  dec_photonPassCleaning(input) = true;
 
+  if(m_egammaAmbiguityTool->accept(input)) dec_photonPassAmbiguity(input) = true;
+
   if (!isAtlfast() && !isData()) {
     if ( m_electronPhotonShowerShapeFudgeTool->applyCorrection(input) != CP::CorrectionCode::Ok)
       ATH_MSG_ERROR("FillPhoton - fudge tool: applyCorrection failed");
