Index: SUSYTools/Root/SUSYObjDef_xAOD.cxx
===================================================================
--- SUSYTools/Root/SUSYObjDef_xAOD.cxx	(revision 758502)
+++ SUSYTools/Root/SUSYObjDef_xAOD.cxx	(working copy)
@@ -99,6 +99,9 @@
 static SG::AuxElement::Decorator<double> dec_effscalefact("effscalefact");
 static SG::AuxElement::Decorator<double> dec_jvtscalefact("jvtscalefact");
 
+static SG::AuxElement::Decorator<char> dec_photonPassCleaning("passCleaning");
+static SG::AuxElement::Decorator<char> dec_photonPassAmbiguity("passAmbiguity");
+
 static SG::AuxElement::Accessor<unsigned int> randomrunnumber("RandomRunNumber");
 
 
@@ -2187,6 +2190,10 @@
   dec_baseline(input) = false;
   dec_isol(input) = false;
 
+  dec_photonPassCleaning(input) = false;
+  dec_photonPassAmbiguity(input) = false;
+  if(m_egammaAmbiguityTool->accept( input )) dec_photonPassAmbiguity(input) = true;
+
   // Author cuts needed according to https://twiki.cern.ch/twiki/bin/view/AtlasProtected/EGammaIdentificationRun2#Photon_authors
   if ( !(input.author() & (xAOD::EgammaParameters::AuthorPhoton + xAOD::EgammaParameters::AuthorAmbiguous)) )
     return StatusCode::SUCCESS;
@@ -2211,6 +2218,7 @@
   float ph_rphi = input.showerShapeValue(xAOD::EgammaParameters::Rphi);
 
   if ( (ph_OQ & 134217728) != 0 && (ph_reta > 0.98 || ph_rphi > 1.0 || (ph_OQ & 67108864) != 0) ) return StatusCode::SUCCESS;
+  dec_photonPassCleaning(input) = true;
 
   if (!isAtlfast() && !isData()) {
     if ( m_electronPhotonShowerShapeFudgeTool->applyCorrection(input) != CP::CorrectionCode::Ok)
@@ -3288,6 +3296,8 @@
 double SUSYObjDef_xAOD::GetMuonTriggerEfficiencySF(const xAOD::Muon& mu, const std::string& trigExpr) {
   
   double eff(1.);
+    #warning setting muon trigger efficiencies to one
+    /*
   int year = treatAsYear();
   if (year == 2015) {
     if (m_muonTriggerSFTool2015->getTriggerEfficiency(mu, eff, trigExpr, m_dataSource) != CP::CorrectionCode::Ok) {
@@ -3302,6 +3312,7 @@
   else {
     ATH_MSG_ERROR("Unknown year returned by treatAsYear() - only have muon trigger SF tools for 2015 and 2016");
   }
+    */
   return eff;
 }
 
@@ -3318,6 +3329,8 @@
     }
   }
 
+    #warning setting trigger SF to one
+    /*
   if (!trigExpr.empty()) {
     if ((trigExpr.find("HLT_2mu14") != std::string::npos) && sfmuons.size() != 2) {
       ATH_MSG_WARNING("Di-muon trigger SF are provided only for events with exaclty two muons!!! Returning 1.");
@@ -3339,6 +3352,7 @@
       }
     }
   }
+    */
   
   return sf;
 }
