Index: SUSYTools/Root/SUSYObjDef_xAOD.cxx
===================================================================
--- SUSYTools/Root/SUSYObjDef_xAOD.cxx	(revision 763336)
+++ SUSYTools/Root/SUSYObjDef_xAOD.cxx	(working copy)
@@ -104,7 +104,10 @@
 
 static SG::AuxElement::Accessor<unsigned int> randomrunnumber("RandomRunNumber");
 
+static SG::AuxElement::Decorator<char> dec_photonPassCleaning("passCleaning");
+static SG::AuxElement::Decorator<char> dec_photonPassAmbiguity("passAmbiguity");
 
+
 SUSYObjDef_xAOD::SUSYObjDef_xAOD( const std::string& name )
   : asg::AsgMetadataTool( name ),
     m_dataSource(Undefined),
@@ -2232,6 +2235,10 @@
   dec_baseline(input) = false;
   dec_isol(input) = false;
 
+  dec_photonPassCleaning(input) = false;
+  dec_photonPassAmbiguity(input) = false;
+  if(m_egammaAmbiguityTool->accept( input )) dec_photonPassAmbiguity(input) = true;
+
   // Author cuts needed according to https://twiki.cern.ch/twiki/bin/view/AtlasProtected/EGammaIdentificationRun2#Photon_authors
   if ( !(input.author() & (xAOD::EgammaParameters::AuthorPhoton + xAOD::EgammaParameters::AuthorAmbiguous)) )
     return StatusCode::SUCCESS;
@@ -2256,6 +2263,7 @@
   float ph_rphi = input.showerShapeValue(xAOD::EgammaParameters::Rphi);
 
   if ( (ph_OQ & 134217728) != 0 && (ph_reta > 0.98 || ph_rphi > 1.0 || (ph_OQ & 67108864) != 0) ) return StatusCode::SUCCESS;
+  dec_photonPassCleaning(input) = true;
 
   if (!isAtlfast() && !isData()) {
     if ( m_electronPhotonShowerShapeFudgeTool->applyCorrection(input) != CP::CorrectionCode::Ok)
