Index: SUSYTools/Root/SUSYObjDef_xAOD.cxx
===================================================================
--- SUSYTools/Root/SUSYObjDef_xAOD.cxx	(revision 723928)
+++ SUSYTools/Root/SUSYObjDef_xAOD.cxx	(working copy)
@@ -1773,8 +1773,9 @@
 
   if (!m_muonSelectionToolBaseline->accept(input)) return StatusCode::SUCCESS;
 
-  if (!(dec_passSignalID(input) = m_muonSelectionTool->accept(input))) return StatusCode::SUCCESS;
+    //if (!(dec_passSignalID(input) = m_muonSelectionTool->accept(input))) return StatusCode::SUCCESS;
 
+
   if (input.pt() <= ptcut || fabs(input.eta()) >= etacut)
     return StatusCode::SUCCESS;
 
@@ -1977,8 +1978,13 @@
 bool SUSYObjDef_xAOD::IsSignalMuon(const xAOD::Muon & input, float ptcut, float d0sigcut, float z0cut)
 {
 
+
   if (!dec_baseline(input)) return false;
 
+  dec_passSignalID(input) = false;
+  if(m_muonSelectionTool->accept(input)) dec_passSignalID(input) = true;
+  if (!dec_passSignalID(input)) return false;
+
   if (input.pt() <= ptcut || input.pt() == 0) return false; // pT cut (might be necessary for leading muon to pass trigger)
   if (z0cut > 0.0 && fabs(dec_z0sinTheta(input)) > z0cut) return false; // longitudinal IP cut
   if (dec_d0sig(input) != 0) {
